src/internal/rest.rs

```diff
@@ -358,77 +358,73 @@ impl<'a> IronCoreRequest<'a> {
+    ) -> Result<B, IronOxideErr> {
         use publicsuffix::IntoUrl;
 
-        let make_req = || {
-            Ok({
-                // build up a request...
-                let mut req = ARequest::new(
-                    Method::POST,
-                    format!("{}{}", self.base_url(), relative_url)
-                        .into_url()
-                        .map_err(|e| IronOxideErr::from((e, error_code)))?,
-                );
-                *req.body_mut() = Some(body.to_vec().into());
-                (req, body.to_vec())
-            })
-        };
-        make_req()
-            .into_future()
-            .and_then(move |(mut req, body_bytes)| {
-                // use the completed request to finish authorization v2 headers
-                let maybe_auth = SignatureUrlString::new(req.url().as_str())
-                    .map(|sig_url| {
-                        auth_b.finish_with(sig_url, req.method().clone(), Some(&body_bytes))
-                    })
-                    .map_err(|e| IronOxideErr::from((e, error_code)));
-
-                // we only support Authorization::Version2 with this call
-                match maybe_auth {
-                    Ok(auth) => {
-                        if let Authorization::Version2 {
-                            user_context,
-                            request_sig,
-                        } = &auth
-                        {
-                            match user_context.to_header(error_code) {
-                                Ok(user_context_header) => {
-                                    replace_headers(req.headers_mut(), user_context_header);
-                                    replace_headers(req.headers_mut(), DEFAULT_HEADERS.clone());
-                                    replace_headers(req.headers_mut(), auth.to_auth_header());
-                                    replace_headers(req.headers_mut(), request_sig.to_header());
-
-                                    Either::A(Self::send_req(req, error_code, move |server_resp| {
-                                        IronCoreRequest::deserialize_body(server_resp, error_code)
-                                    }))
-                                }
-                                Err(e) => Either::B(futures::future::err(e)),
-                            }
-                        } else {
-                            panic!("authorized requests must use version 2 of API authentication")
+        let make_req: Result<_, IronOxideErr> = Ok({
+            // build up a request...
+            let mut req = ARequest::new(
+                Method::POST,
+                format!("{}{}", self.base_url(), relative_url)
+                    .into_url()
+                    .map_err(|e| IronOxideErr::from((e, error_code)))?,
+            );
+            *req.body_mut() = Some(body.to_vec().into());
+            (req, body.to_vec())
+        });
+        let (mut req, body_bytes) = make_req?; //COLT: See if you can do better
```

Comment:
Is this comment to yourself for later or have you already tried to do better and forgot to remove it?

Reply:
Removed.